includes/db/Database.php

```diff
@@ -876,6 +915,8 @@ public function query( $sql, $fname = '', $tempIgnore = false ) {
 
 		$this->mLastQuery = $sql;
 		$is_writeable = $this->isWriteQuery( $sql );
+		$isNonTempWrite = !$this->registerTempTableOperation( $sql );
```

Comment:
I would rather see it named `$isTempDB` or similar which is basically what this function determines. For me it would be clearer.